Sources/FoundationEssentials/URL/URL.swift

```diff
@@ -2200,16 +2210,29 @@ extension URL {
 #if !NO_FILESYSTEM
     private static func isDirectory(_ path: String) -> Bool {
         #if os(Windows)
+        guard !path.isEmpty else { return false }
         let path = path.replacing(._slash, with: ._backslash)
-        #endif
-        #if !FOUNDATION_FRAMEWORK
-        var isDirectory: Bool = false
-        _ = FileManager.default.fileExists(atPath: path, isDirectory: &isDirectory)
-        return isDirectory
+        return (try? path.withNTPathRepresentation { pwszPath in
+            // If path points to a symlink (reparse point), get a handle to
+            // the symlink itself using FILE_FLAG_OPEN_REPARSE_POINT.
+            let handle = CreateFileW(pwszPath, GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nil, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT, nil)
+            guard handle != INVALID_HANDLE_VALUE else { return false }
+            defer { CloseHandle(handle) }
+            var info: BY_HANDLE_FILE_INFORMATION = BY_HANDLE_FILE_INFORMATION()
+            guard GetFileInformationByHandle(handle, &info) else { return false }
+            if (info.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) == FILE_ATTRIBUTE_REPARSE_POINT { return false }
+            return (info.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) == FILE_ATTRIBUTE_DIRECTORY
```

Comment:
Hi @compnerd, is creating the file with `FILE_FLAG_OPEN_REPARSE_POINT` and then checking the attributes for `FILE_ATTRIBUTE_REPARSE_POINT` the correct way to achieve these semantics on Windows: "return false if the path ends in a symlink, otherwise return true if the path ends at a directory"?

Reply:
Yeah, I think that this is the correct way. You need to check the `FILE_ATTRIBUTE_REPARSE_POINT` first as `FIEL_TYPE_DIRECTORY` can also be set to indicate a junction.